src/cliffordsynthesis/encoding/PhaseCorrectionEncoder.cpp

```diff
@@ -0,0 +1,148 @@
+#include "cliffordsynthesis/encoding/PhaseCorrectionEncoder.hpp"
+
+#include "LogicTerm/Logic.hpp"
+#include "LogicTerm/LogicTerm.hpp"
+#include "cliffordsynthesis/GateSet.hpp"
+#include "cliffordsynthesis/Utils.hpp"
+#include "utils/logging.hpp"
+
+#include <bitset>
+#include <cstddef>
+#include <string>
+#include <vector>
+
+namespace cs::encoding {
+using namespace logicbase;
+
+std::vector<qc::OpType> PhaseCorrectionEncoder::phaseCorrection() {
+  createVariables();
+  encodePauliConstraints();
+  lb->solve();
+  return extractResult();
+}
+
+void PhaseCorrectionEncoder::createVariables() {
+  DEBUG() << "Creating phase correction gate variables.";
+  for (std::size_t i = 0U; i < N; ++i) {
+    paulis[i].reserve(4);
+    for (const auto& pauli : PAULIS) {
+      const std::string gName =
+          "g_" + std::to_string(i) + "_" + toString(pauli);
+      TRACE() << "Creating pauli variable " << gName;
+      paulis[i].emplace_back(lb->makeVariable(gName));
+    }
+  }
+
+  const std::bitset<64> init = uncorrected.getBVFrom(2 * N);
+  initialPhase               = vectorFromBitset(init);
+  const std::bitset<64> tar  = target.getBVFrom(2 * N);
+  targetPhase                = vectorFromBitset(tar);
+}
+
+void PhaseCorrectionEncoder::encodePauliConstraints() {
+  DEBUG() << "Encoding pauli constraints.";
+  splitXorR(initialPhase);
+  for (std::size_t q = 0U; q < N; ++q) {
+    const auto& xCol = vectorFromBitset(uncorrected.getBVFrom(q));
+    const auto& zCol = vectorFromBitset(uncorrected.getBVFrom(N + q));
+
+    LogicVector iChange{};
+    LogicVector xChange{};
+    LogicVector zChange{};
+    LogicVector yChange{};
+    iChange.reserve(S);
+    xChange.reserve(S);
+    zChange.reserve(S);
+    yChange.reserve(S);
+
+    for (std::size_t row = 0U; row < S; ++row) {
+      DEBUG() << "Encoding identity constraint.";
+      iChange.emplace_back(logicbase::LogicTerm::ite(
+          paulis[q][0], LogicTerm(false), LogicTerm(false)));
```

Comment:
Something seems off to me about that constraint no matter the value of the Paulis, this will always be false, right?
Like, this can be way simpler and outside the loop.